### ISSUE TITLE
json_object_object_foreach warnings

### ISSUE THREAD

**Issue by ghazel** [json-c/json-c#188]
Clang spits out two warnings:

```c
json_object_object_foreach(o, key, value) {
  printf("%p %p\n", key, value);
}
```

```
warning: variable 'key' may be uninitialized when used here [-Wconditional-uninitialized]
warning: variable 'value' may be uninitialized when used here [-Wconditional-uninitialized]
```

**Comment by ghazel** [comment]
Seems like a spurious warning, since the condition which sets `key` and `val` would also break the loop if it weren't true. However, it's fairly easily silenced by initializing key and val in the macro:

```c
# define json_object_object_foreach(obj,key,val) \
	char *key __attribute__((__unused__)) = NULL; \
	struct json_object *val __attribute__((__unused__)) = NULL; \
	for(struct lh_entry *entry ## key = json_object_get_object(obj)->head, *entry_next ## key = NULL; \
		({ if(entry ## key) { \
			key = (char*)entry ## key->k; \
			val = (struct json_object*)entry ## key->v; \
			entry_next ## key = entry ## key->next; \
		} ; entry ## key; }); \
		entry ## key = entry_next ## key )
```

**Comment by hawicz** [comment]
Looks like this was already fixed, probably by changeset 75d7409c.
